internal/routing/handlers/user_handler.go

```diff
@@ -437,3 +443,308 @@ func generateToken() string {
+			utils.WriteAndLogError(w, schemas.SubscriptionNotFound, http.StatusNotFound, errors.New("subscription not found"))
+			return
+		}
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Commit the transaction
+	if err := utils.CommitTransaction(w, tx, transactionCtx, cancel); err != nil {
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// SearchUsers returns a list of users that match the search query using query parameters using offset and limit.
+func (handler *UserHandler) SearchUsers(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithDeadline(r.Context(), time.Now().Add(10*time.Second))
+	defer func() {
+		if err := ctx.Err(); err != nil {
+			log.Debug("Context error: ", err)
+		}
+		cancel()
+		log.Debug("Context canceled")
+	}()
+
+	// Get the search query from the query parameters
+	searchQuery := r.URL.Query().Get("username")
+
+	// Get the offset from the query parameters
+	offsetString := r.URL.Query().Get("offset")
+	if offsetString == "" {
+		offsetString = "0"
+	}
+	offset, err := strconv.Atoi(offsetString)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("offset invalid"))
+		return
+	}
+
+	// Get the limit from the query parameters
+	limitString := r.URL.Query().Get("limit")
+	if limitString == "" {
+		limitString = "10"
+	}
+	limit, err := strconv.Atoi(limitString)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("limit invalid"))
+		return
+	}
+
+	// Get the length of all users that match the search query
+	queryString := "SELECT COUNT(*) FROM alpha_schema.users"
+	row := handler.DatabaseManager.GetPool().QueryRow(ctx, queryString)
+	var records int
+	if err := row.Scan(&records); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Check if the offset is valid
+	if offset > records {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("offset greater than records"))
+		return
+	}
+
+	// Get the users that match the search query
+	queryString = "SELECT username, nickname, profile_picture_url, levenshtein(username, $1) as ld FROM alpha_schema.users ORDER BY ld LIMIT $2 OFFSET $3"
+	rows, err := handler.DatabaseManager.GetPool().Query(ctx, queryString, searchQuery, limit, offset)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+	defer rows.Close()
+
+	// Create a list of users
+	users := make([]schemas.UserSearchDTO, 0)
+
+	for rows.Next() {
+		user := schemas.UserSearchDTO{}
+		if err := rows.Scan(&user.Username, &user.Nickname, &user.ProfilePictureURL, &user.LevenshteinDistance); err != nil {
+			utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+			return
+		}
+		users = append(users, user)
+	}
```

Comment:
Dementsprechend brauchen wir auch nicht alle zutreffenden Records hier, sondern nur von `[offset, offset + limit]`

Reply:
Der offset und das limit sind eig schon in der sql query angegeben

Reply:
Aber das wollen wir doch nicht, oder? Ich konsolidier mal die und die Konversation unter der, da es das gleiche Thema behandelt.

Was wäre denn eine sinnvolle Levenshtein-Distanz, auf die man sich festlegen kann? Dann kann man das Limit aus der SQL-Query rausnehmen und halt wie eigentlich vorgedacht, `records` richtig befüllen mit der tatsächlichen Anzahl an Treffern.

Reply:
### Bezüglich offset/limit in der Query:
Es kommt halt alles auf unsere Erwartungen an. Ich habe jetzt es hier einfach gehalten und gebe vorerst alle User zurück ohne eingrenzung aber limit und offset impliziert in der SQL-Query, weil wir eh nicht mehr als 10 User oder so in der Datenbank haben werden.

Alternativ könnten wir die levenshtein distanz in der Query cappen und dann den offset und limit später anwenden. Ich denke nur dass die beiden eine gleiche Performance haben weil du trotzdem über alle User in der Datenbank eine Levenshtein Distanz ausrechnen musst, bevor du sie dementsprechend. 

### Bezüglich dem Cap:
Ich denke ein Cap von 5 für Usernamen passt aufgrund der kleinen Anzahl an Usern. Wenn du willst kann ich das noch einbauen.